internal/app/networkd/pkg/nic/nic.go

```diff
@@ -306,18 +306,26 @@ func (n *NetworkInterface) Renew() {
 // address TTL. If that fails, we'll continue to attempt to retry every
 // halflife.
 func (n *NetworkInterface) renew(method address.Addressing) {
+	const minRenewDuration = 5 * time.Second // protect from renewing too often
+
 	renewDuration := method.TTL() / 2
 
 	var err error
 
 	for {
-		<-time.After(renewDuration)
+		time.Sleep(renewDuration)
 
 		if err = n.configureInterface(method, n.Link); err != nil {
+			log.Printf("failure to renew address for %q: %s", n.Name, err)
+
 			renewDuration = (renewDuration / 2)
```

Comment:
Why do we even _have_ this ever-shrinking renew duration?  Why should this be any different from minRenewDuration at all?

Reply:
I don't have any good idea, this looks like weird retry mechanism. I think the idea is that `1/2 + 1/4 + 1/8 + ... < 1`

Reply:
exactly; this progression doesn't make sense to me